spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -1224,6 +1222,14 @@ struct ProductionVisitor : public production::Visitor {
         else if ( const auto& unit = p->tryAs<production::Unit>() )
             unit_type = unit->unitType();
 
+        // Call any `%sync_advance` hook as we enter sync mode.
```

Comment:
This is still only WIP. We still call the hook twice when entering for some cases, e.g., the setups in `spicy.types.unit.sync-advance` where we invoke it both when entering and at the same offset since we now jumped across a gap. I tried adding special handling to suppress the second case by inspecting `search_start`, but we do not set this variable (in the same scope) for all synchronization scenarios.

Reply:
Fixed.